config/v1/types_oauth.go

```diff
@@ -10,18 +10,22 @@ import (
 // +genclient:nonNamespaced
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 
-// OAuth holds cluster-wide information about OAuth.  The canonical name is `cluster`
+// OAuth holds cluster-wide information about OAuth.  The canonical name is `cluster`.
+// It is used to configure the integrated OAuth server.
+// This configuration is only honored when the top level Authentication config has type set to IntegratedOAuth.
 type OAuth struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata"`
 	Spec              OAuthSpec   `json:"spec"`
-	Status            OAuthStatus `json:"status,omitempty"`
+	Status            OAuthStatus `json:"status"`
```

Comment:
Why drop `omitempty`?  I'd expect folks using this `type` to generate configurations would want to set a dummy `status`, but maybe folks generating configurations will just be using `OAuthSpec` and not `OAuth`?  The current repository (including `vendor/`) seems conflicted about this (or my `grep` is too rough and is picking up other things too):

```console
$ git describe --always --abbrev=40
4703f3e71d833812a81e0fa3a2da1257e3efd85c
$ git grep '^[[:space:]]*Status[[:space:]]' | grep omitempty | wc -l
69
$ git grep '^[[:space:]]*Status[[:space:]]' | grep -v omitempty | wc -l
59
```

Reply:
I was just going by the other `status` fields in the config v1 package.  No strong opinion either way.